source/develop/release-management/features/storage/pass_discard_from_guest_to_underlying_storage.html.md

```diff
@@ -0,0 +1,234 @@
+When the free space is about to end, we extend the lv with one more Gigabyte, so that the disk grows again and again until it reaches its virtual size.<br/>
+If the lun that is used to create the storage domain is also thin provisioned, it behaves in a similar way. That is, when we extend our disk with one Gigabyte, we don't necessarily get 1G from the storage, but we do get 1G of virtual space.
+
+Similarly, in file domains the underlying thin provisioned lun gets bigger as the corresponding disk's file grows.
+
+### What we would like to achieve
+When files are removed from the disk by the guest, the lv does not shrink (it would if we called `lvreduce`, but we don't call it for technical reasons). Since no change was made to the size of the lv, the thin provisioned lun in the underlying storage haven't changed either.
+So in fact, although we freed some of the disk's space, the underlying storage is not aware of it, and practically we got a situation where the underlying thin provisioned luns can only grow, and can never reduce.
+
+The same goes with file storage - when files are removed by the guest, the disk's actual size remains the same and so is the size of the underlying thin provisioned luns.
+
+A better approach would be to "tell" the underlying storage that we freed some space that we don't need anymore, so that someone else can use it.
+
+### For whom is this feature useful
+This feature is pretty useful for anyone who uses thin provisioned disks and wants to utilize them more efficiently.
+
+
+## Owner
+
+* Idan Shaby
+* Email: <ishaby@redhat.com>
+
+
+## Current status
+
+* engine - work in progress<br/>
+    * For block storage, the engine should query vdsm about the storage's `discard_max_bytes` and `discard_zeroes_data` values. Then it will decide whether to allow enabling *Pass Discard* or not.<br/>
+    For more information see [General Functionality and Restrictions](#general-functionality-and-restrictions).
+    * As discarding works also on file storage, the engine should not block the user from using it.
+    * A few checks should be added to flows like moving or copying a disk from file to block storage with *Pass Discard* and *Wipe After Delete* enabled.
+    * Decide about the relationship between *Pass Discard* and *Enable SCSI Pass-Through* for direct LUNs - it should be discussed with the UX team.
+* api-model - released in 4.1.1
+* vdsm - work in progress
+
+
+## General functionality and restrictions
+
+### Under the hood
+There's a new virtual machine's disk property called ***"Pass Discard"***.<br/>
+If all the [restrictions](#restrictions) are met and *"Pass Discard"* is enabled, when a discard command (UNMAP SCSI command) is sent from the guest, the VM will not ignore it and will pass it on to the underlying storage. Then, the storage will set the unused blocks as free so that others can use them, and the size of the underlying thin provisioned lun will reduce.
+
+### Who can consume the discarded blocks
```

Comment:
People cannot consume discarded blocks, so "who" is the wrong pronoun to use here.